src/data_readers/data_reader_jag_conduit.cpp

```diff
@@ -526,6 +551,95 @@ void data_reader_jag_conduit::check_input_keys() {
 
 
 #ifndef _JAG_OFFLINE_TOOL_MODE_
+void data_reader_jag_conduit::determine_num_samples_to_use() {
+#if 1
+  // The meaning of m_first_n is slightly different in this data reader as it
+  // represents the first n local samples instead of the first n global samples.
+  if (m_first_n > 0) {
+    const size_t num_samples = std::min(static_cast<size_t>(m_first_n), get_num_valid_local_samples());
+    m_valid_samples.resize(num_samples); // this does not work with unordered_map but with vector
+  }
+#else
+  if (m_first_n > 0) {
+    _THROW_LBANN_EXCEPTION_(_CN_, "load() does not support first_n feature.");
+  }
+#endif
+
+#if 1
+  // We do not support "percent_of_data_to_use" or "absolute_sample_count" yet.
+  if ((get_use_percent() != 1.0) || (get_absolute_sample_count() != static_cast<size_t>(0u))) {
+    _THROW_LBANN_EXCEPTION_(get_type(), \
+      "'percent_of_data_to_use' and 'absolute_sample_count' are not supported with this data reader");
+  }
+  if (get_validation_percent() != 0.0) {
+    _THROW_LBANN_EXCEPTION_(get_type(), \
+      "'validation_percent' is not supported with this data reader");
+  }
+#else
+  select_subset_of_data();
+#endif
+
+  const size_t num_valid_samples = get_num_valid_local_samples();
+
+  MPI_Comm comm = m_comm->get_model_comm().comm;
+  const size_t my_rank = static_cast<size_t>(m_comm->get_rank_in_model());
+  const size_t num_ranks = static_cast<size_t>(m_comm->get_procs_per_model());
+
+  // Find the minimum of the number of valid samples locally available
+  unsigned long long n_loc = static_cast<unsigned long long>(num_valid_samples);
+  unsigned long long n_min = static_cast<unsigned long long>(num_valid_samples);
+  MPI_Allreduce(&n_loc, &n_min, 1, MPI_UNSIGNED_LONG_LONG, MPI_MIN, comm);
```

Comment:
You could use: template <typename T>
  T allreduce(T snd, const El::mpi::Comm c, El::mpi::Op op = El::mpi::SUM)

Reply:
Thanks. I was looking for this.